ganttproject/src/net/sourceforge/ganttproject/ResourceTreeTable.java

```diff
@@ -123,12 +131,93 @@ protected Chart getChart() {
+      return o2.getName().compareTo(o1.getName());
+    }
+  }
+
+
   /** Initialize the treetable. Addition of various listeners, tree's icons. */
   @Override
   protected void doInit() {
     super.doInit();
     myResourceTreeModel.updateResources();
+    getTable().getModel().addTableModelListener(new ModelListener());
     getVerticalScrollBar().addAdjustmentListener(new VscrollAdjustmentListener(myUiFacade.getResourceChart(), false));
+    getTableHeader().addMouseListener(new MouseAdapter() {
+      @Override
+      public void mouseClicked(MouseEvent mouseEvent) {
+        int index = getTable().columnAtPoint(mouseEvent.getPoint());
+        if (index == -1){
+          return;
+        }
+
+        if (mouseEvent.isPopupTrigger() || mouseEvent.getButton() != MouseEvent.BUTTON1){
+          return;
+        }
+
+        if (mouseEvent.isAltDown() || mouseEvent.isShiftDown() || mouseEvent.isControlDown()){
+          return;
+        }
+
+        final TableHeaderUiFacadeImpl tableHeader = getTableHeaderUiFacade();
+        final ColumnImpl column = tableHeader.findColumnByViewIndex(index);
+        final ResourceDefaultColumn resourceColumn = ResourceDefaultColumn.find(column.getID());
+
+        getUiFacade().getUndoManager().undoableEdit(GanttLanguage.getInstance().getText("task.sort"), new Runnable() {
+          @Override
+          public void run() {
+
+            if (resourceColumn == ResourceDefaultColumn.NAME) {
+
+              if (column.getSort() == SortOrder.ASCENDING) {
+                column.setSort(SortOrder.DESCENDING);
+                resourceColumn.setSortComparator(new DescendingNameComparator());
+              } else {
+                column.setSort(SortOrder.ASCENDING);
+                resourceColumn.setSortComparator(new AscendingNameComparator());
+              }
+
+              ArrayList<HumanResource> sorted = new ArrayList<>();
+              getProject().getHumanResourceManager().getResources().forEach( h -> {
+                sorted.add(h);
+              });
+              sorted.sort((Comparator<HumanResource>) resourceColumn.getSortComparator());
+
+              for(int j = sorted.size()-1; j>=0; j--){
+                HumanResource h = sorted.get(j);
+                int idxo =  getProject().getHumanResourceManager().getResources().indexOf(h);
+                int idxs = sorted.indexOf(h);
+                int delta = idxs - idxo;
+                for(int i = 0; i < Math.abs(delta) ; i++){
+                  if (idxs < idxo) {
+                    myResourceTreeModel.moveUp(h);
+                  }  else {
+                    myResourceTreeModel.moveDown(h);
+                  }
+                }
+              };
+            }
+          }
+        });
+      }
+    });
+  }
+
+  private class ModelListener implements TableModelListener {
+    @Override
+    public void tableChanged(TableModelEvent e) {
+      getUiFacade().getGanttChart().reset();
```

Comment:
copy'n'paste failure, we're not in gantt chart here. Please refactor this too, so that there was common `ModelListener` in the base class.